ui/components/app/snaps/update-snap-permission-list/update-snap-permission-list.js

```diff
@@ -13,40 +15,48 @@ export default function UpdateSnapPermissionList({
 }) {
   const t = useI18nContext();
 
+  const { name: snapName } = useSelector((state) =>
+    getSnapMetadata(state, targetSubjectMetadata.origin),
+  );
+
   return (
     <Box paddingTop={1}>
-      {getWeightedPermissions(t, newPermissions, targetSubjectMetadata).map(
-        (permission, index) => (
-          <PermissionCell
-            permissionName={permission.permissionName}
-            title={permission.label}
-            description={permission.description}
-            weight={permission.weight}
-            avatarIcon={permission.leftIcon}
-            dateApproved={permission?.permissionValue?.date}
-            key={`${permission.permissionName}-${index}`}
-          />
-        ),
-      )}
-      {getWeightedPermissions(t, revokedPermissions, targetSubjectMetadata).map(
-        (permission, index) => (
-          <PermissionCell
-            permissionName={permission.permissionName}
-            title={permission.label}
-            description={permission.description}
-            weight={permission.weight}
-            avatarIcon={permission.leftIcon}
-            dateApproved={permission?.permissionValue?.date}
-            key={`${permission.permissionName}-${index}`}
-            revoked
-          />
-        ),
-      )}
-      {getWeightedPermissions(
+      {getWeightedPermissions({
+        t,
+        permissions: newPermissions,
```

Comment:
We need to pass `getSubjectName` to all instances of `getWeightedPermissions`

Reply:
Why? we're providing the `subjectName` already. The `getSubjectName` function is only really used by the `wallet_snap` permission.

Reply:
Yeah, but the `wallet_snap` permission can be used everywhere right?

Reply:
oh right, snap dependencies. I'll fix.